addons/mrp/models/product.py

```diff
@@ -30,6 +30,14 @@ def _compute_used_in_bom_count(self):
             template.used_in_bom_count = self.env['mrp.bom'].search_count(
                 [('bom_line_ids.product_id', 'in', template.product_variant_ids.ids)])
 
+    def write(self, values):
+        super().write(values)
+        if 'active' in values:
+            if values['active']:
+                self.with_context(active_test=False).bom_ids.active = True
+            else:
+                self.bom_ids.active = False
```

Comment:
@oma-odoo Should we remove the BoM line that contains the archived product since they couldn't be archive or let the user do it?
cc @yhu-odoo

Reply:
@amoyaux I wouldn't archive them automatically because then the BOM's will still be usable and it will be complex for the end user to realize that maybe he should have replaced one product which isn't used anymore (as it would have simply disappeared).  Thanks
cc @yhu-odoo